chain-impl-mockchain/src/rewards.rs

```diff
@@ -163,6 +174,24 @@ pub fn rewards_contribution_calculation(epoch: Epoch, params: &Parameters) -> Va
     }
 }
 
+/// Added new function to ensure rewards calculation complies with research -- KH
+/// Scale the total rewards in proportion to the total delegated stake
+pub fn rewards_contribution_calculation_scaled(
+    epoch: Epoch,
+    params: &Parameters,
+    total_delegated: Stake,
+) -> Value {
+    let r = rewards_contribution_calculation(epoch, params);
+
+    // limit the rewards if necessary
+    let rl = &params.rewards_limit;
+    const SCALE: u128 = 10 ^ 18;
+
+    let limit_scaled = u64::from(total_delegated) as u128 * SCALE;
+    let limit = limit_scaled / rl.denominator.get() as u128;
+    cmp::min(Value((limit / SCALE) as u64), r)
+}
+
```

Comment:
doesn't really make sense to me. shouldn't this be the ratio of delegated stake against the potential total stake ?

Reply:
You don't need to know the total potential stake (which might anyway need a new fixed constant parameter to limit the system).  The rewards can be scaled against the delegated stake.